packages/jsonforms-zorro-wrapper/src/controls/wysiwyg.renderer.ts

```diff
@@ -21,19 +22,39 @@ import { NzValidationStatusPipe } from '../other/validation-status.pipe';
         </nz-form-label>
       }
       <DescriptionRenderer [uiSchema]="uischema" [scopedSchema]="scopedSchema"></DescriptionRenderer>
-      <nz-form-control [nzHasFeedback]="showValidationStatus" [nzErrorTip]="errorMessage" [nzValidateStatus]="form.status | nzValidationStatus">
-        <quill-editor
-          [id]="id"
-          [formControl]="form"
-          [modules]="editorModules"
-          [formats]="formats"
-          linkPlaceholder="Paste your link here"
-          [placeholder]="placeholder"
-          (ngModelChange)="onChange($event)"
-          [styles]="{ minHeight: '100px' }"
-          (blur)="triggerValidation()"
-        ></quill-editor>
-      </nz-form-control>
+      @if (hasExternalValidation) {
```

Comment:
[optional]

This duplicates the code. Have you thought about adding conditions inline? I can see that only `[nzErrorTip]` and `[nzValidateStatus]` has changed.

```suggestion
      <nz-form-control
        [nzHasFeedback]="showValidationStatus"
        [nzErrorTip]="hasExternalValidation ? stringFieldErrorMessages.join(', ') : errorMessage"
        [nzValidateStatus]="hasExternalValidation ? (stringFieldErrorMessages.length ? 'error' : 'success') : (form.status | nzValidationStatus)" 
      >
        <quill-editor
          [id]="id"
          [formControl]="form"
          [modules]="editorModules"
          [formats]="formats"
          linkPlaceholder="Paste your link here"
          [placeholder]="placeholder"
          (onContentChanged)="onChange($event)"
          [styles]="{ minHeight: '100px' }"
          (blur)="triggerValidation()"
        ></quill-editor>
      </nz-form-control>
```